src/BuilderMacros.php

```diff
@@ -0,0 +1,148 @@
+         *
+         * @return int
+         */
+        return function (): int {
+            $raw = $this->engine()->search($this);
+
+            return array_key_exists('nbHits', $raw) ? (int) $raw['nbHits'] : 0;
+        };
+    }
+
+    /**
+     * @return \Closure
+     */
+    public function aroundLatLng(): Closure
+    {
+        /**
+         * Search for entries around a given location.
+         *
+         * @link https://www.algolia.com/doc/guides/geo-search/geo-search-overview/
+         *
+         * @param float $lat Latitude of the center
+         * @param float $lng Longitude of the center
+         *
+         * @return \Laravel\Scout\Builder
+         */
+        return function (float $lat, float $lng): Builder {
+            $callback = $this->callback;
+            $this->callback = function ($algolia, $query, $options) use ($lat, $lng, $callback) {
+                $options['aroundLatLng'] = (float) $lat.','.(float) $lng;
+                if ($callback) {
+                    return call_user_func($callback, $algolia, $query, $options);
+                }
+
+                return $algolia->search($query, $options);
+            };
+
+            return $this;
+        };
+    }
+
+    /**
+     * @return \Closure
+     */
+    public function with(): Closure
+    {
+        /**
+         * Customize the request with the provided options
+         *
+         * @param array $options Latitude of the center
+         *
+         * @return \Laravel\Scout\Builder
+         */
+        return function (array $options): Builder {
+            $callback = $this->callback;
+
+            $this->callback = function ($algolia, $query, $baseOptions) use ($options, $callback) {
+                $options = array_merge($options, $baseOptions);
+
+                if ($callback) {
+                    return call_user_func($callback, $algolia, $query, $options);
+                }
+
+                return $algolia->search($query, $options);
+            };
+
+            return $this;
+        };
+    }
+
+    /**
+     * @return \Closure
+     */
+    public function hydrate(): Closure
+    {
+        /**
+         * Create a collection of models from search results.
+         *
+         * @return \Illuminate\Support\Collection
+         */
+        return function () {
+            $results = $this->engine()->search($this);
+
+            $models = collect();
+
+            if (count($results['hits']) === 0) {
+                return $models;
+            }
+
+            $hits = collect($results['hits']);
+            $className = get_class($this->model);
+
+            /**
+             * If the model is fully guarded, we unguard it. Fully guarded is the default
+             * configuration and it will result in error. If the `$guarded` attribute
+             * exists on the model class, we will take it in consideration.
+             *
+             * @todo Review this algorithm.
```

Comment:
I forgot about the `hydrate` method 😄  Did you get a chance to review it?

Reply:
@julienbourdeau Not yet. This `@todo` is exactly for that. To later review algorithm, maybe it is already good. 🙏